### DISCUSSION TITLE
Plotman plot not seeing running jobs in Linux VM on Windows

### DISCUSSION THREAD

**Discussion by zorner** [guydavis/machinaris#815] [Q&A]
#814 details how I have gotten 'plotman archive' working in a Linux VM on Windows. 

'plotman plot' will start fine, but as soon as the global_stagger_m times out, it starts another plot plotting. Left unattended, the tmp dir fills up and all plotting fails. 'plotman archive' also detects archive jobs, but seams to have no issues.

Machinaris using the same plotman.yaml knows there are active plotting jobs and waits after the global_stagger_m times out until a plotting job gets to stage 5.

While I have made the Linux VM see everything the same as the docker container using sym links, there are still differences.

- Machinaris dose not have a /root/.config dir, so plotman.yaml can not be in /root/.config. It is in /root/.chia/plotman.
- VM has /root/.config/plotman.yaml as a sym link to /root/.chia/plotman/plotman.yaml.
- VM comes with python 3.8. Machinaris uses 3.10. I have tried 3.10 and 3.9 in the VM, but did not make a difference.
- VM may not have python3.??-distutils and python3.??-dev installed. I have tried with them, but no change.
- VM only has the Chia CLI. Machinaris may have the Chia GUI version.

Any ideas what is missing? I am not sure what to try next.

**Comment by guydavis** [comment]
Interesting setup of Chia + Plotman on Windows inside a Linux VM.  Sorry, I only have time to support Machinaris (Docker) however.  Good luck with your explorations!

**Comment by zorner** [comment]
When trying to run both 'plotman plot' and 'plotman archive' at the same time, after some time (140s - 1780s) it errors out:
```
Traceback (most recent call last):
  File "/opt/chia/venv/bin/plotman", line 8, in <module>
    sys.exit(main())
  File "/opt/chia/venv/lib/python3.8/site-packages/plotman/plotman.py", line 281, in main
    (started, msg) = manager.maybe_start_new_plot(
  File "/opt/chia/venv/lib/python3.8/site-packages/plotman/manager.py", line 118, in maybe_start_new_plot
    jobs = job.Job.get_running_jobs(log_cfg.plots)
  File "/opt/chia/venv/lib/python3.8/site-packages/plotman/job.py", line 187, in get_running_jobs
    job = cls(
  File "/opt/chia/venv/lib/python3.8/site-packages/plotman/job.py", line 215, in __init__
    for f in self.proc.open_files():
  File "/opt/chia/venv/lib/python3.8/site-packages/psutil/__init__.py", line 1142, in open_files
    return self._proc.open_files()
  File "/opt/chia/venv/lib/python3.8/site-packages/psutil/_pslinux.py", line 1645, in wrapper
    return fun(self, *args, **kwargs)
  File "/opt/chia/venv/lib/python3.8/site-packages/psutil/_pslinux.py", line 2193, in open_files
    path = readlink(file)
  File "/opt/chia/venv/lib/python3.8/site-packages/psutil/_pslinux.py", line 210, in readlink
    path = os.readlink(path)
OSError: [Errno 9] Bad file descriptor: '/proc/6620/fd/174'

[1]-  Exit 1                  plotman plot
[2]+  Exit 1                  plotman archive
```
/proc/6620 is the actively running plotting process:
```
root@Plotter2-Chia:/mnt/c/Users/Cryptnoid Pennies# ps -x | grep "plotman\|rsync\|chia_plot"
 6620 ?        SNsl 535:52 chia_plot -n 1 -r 24 -u 256 -t /plotting1/ -d /plotting1/ -2 /ramdrive/ -v 256 -K 1 -f a3f7e719a508879c16144b615e76aeab1c2bea66b47b690aea33c9e5fed591ad4476b573f2e3dddfe84e923adeae6449 -c xch1ekslymyyfrrv50nptterxuxgmwvw0vc8xz848sf3gzfhse7e8thqmjlqj4
 8220 tty2     S      0:00 grep --color=auto plotman\|rsync\|chia_plot
```
As this happens randomly, it suggest this same segment of code has run successfully several times before having an issue.

A quick search online suggests the python scripts might not be conforming to standard syntax. I doubt this, as these scripts have been being used for over a year. I know very little python.

It feels more like some lib or some python support programs are missing or need updating.
